server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1773,39 +1782,121 @@ public File exportData(
+        catch (ExportCreationException e) {
+            throw new IseException(i18n.tr("Unable to create export archive"), e);
         }
+    }
 
-        poolManager.regenerateDirtyEntitlements(entitlementCurator.listByConsumer(consumer));
+    /**
+     * Initiates an async generation of a compressed file representation of a {@link Consumer} (manifest).
+     * The response will contain the id of the job from which its result data will contain the href to
+     * download the generated file.
+     *
+     * @param response
+     * @param consumerUuid the uuid of the target consumer.
+     * @param cdnLabel
+     * @param webAppPrefix
+     * @param apiUrl
+     * @return the details of the async export job that is to be started.
+     */
+    @ApiOperation(
+        notes = "Initiates an async generation of a Compressed File representation of a Consumer " +
+        "(manifest). The response will contain the id of the job from which its result data " +
+        " will contain the href to download the generated file.",
+        value = "Async Consumer Export (manifest)",
+        response = JobDetail.class)
+    @ApiResponses({ @ApiResponse(code = 403, message = ""), @ApiResponse(code = 500, message = ""),
+        @ApiResponse(code = 404, message = "") })
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{consumer_uuid}/export/async")
+    public JobDetail exportDataAsync(
+        @Context HttpServletResponse response,
+        @PathParam("consumer_uuid") @Verify(Consumer.class) String consumerUuid,
+        @QueryParam("cdn_label") String cdnLabel,
+        @QueryParam("webapp_prefix") String webAppPrefix,
+        @QueryParam("api_url") String apiUrl) {
+        Consumer consumer = verifyExportParams(consumerUuid, cdnLabel);
+        return manifestManager.generateManifestAsync(consumer, cdnLabel, webAppPrefix, apiUrl);
+    }
 
-        File archive;
-        try {
-            archive = exporter.getFullExport(consumer, cdnLabel, webAppPrefix, apiUrl);
-            response.addHeader("Content-Disposition", "attachment; filename=" +
-                archive.getName());
+    /**
+     * Downloads an asynchronously generated consumer export file (manifest). If the file
+     * was successfully downloaded, it will be deleted.
+     *
+     * @param response
+     * @param consumerUuid the UUID of the target consumer.
+     * @param exportId the id of the stored export.
+     */
+    @ApiOperation(
+        notes = "Downloads an asynchronously generated consumer export file (manifest).",
+        value = "Async Consumer Export (manifest) Download",
+        response = File.class)
+    @ApiResponses({ @ApiResponse(code = 403, message = ""), @ApiResponse(code = 500, message = ""),
+        @ApiResponse(code = 404, message = "") })
+    @GET
+    @Produces("application/zip")
+    @Path("{consumer_uuid}/export/{export_id}")
+    public void downloadExistingExport(
```

Comment:
> Does it make sense to provide a query param that allows us to skip the delete? ( for debugging purposes )
> 
> if not, and I am sure we probably discussed this earlier in emails, but is this not a DELETE instead of a GET?
> I understand that the intent is to download, but using delete clarifies to the client that this will delete as well.

I'm not sure that giving the caller the opportunity to not delete would be a good idea. Especially since we are trying to prevent DB swell. That being said, I think that the whole debug/delete issue may be worth a conversation with the team. I'll bring it up next scrum. 

Personally I think that the request is a GET. The file deletion in candlepin is just implementation detail that should be documented. i.e The caller isn't making the request to DELETE a manifest file.